modules/core/src/main/scala/org/scalasteward/core/data/Dependency.scala

```diff
@@ -30,10 +30,10 @@ final case class Dependency(
     configurations: Option[String] = None
 ) {
   def formatAsModuleId: String =
-    s""""$groupId" % "$artifactIdCross" % "$version""""
-
-  def formatAsModuleIdScalaVersionAgnostic: String =
-    s""""$groupId" %% "$artifactId" % "$version""""
+    artifactId match {
+      case `artifactIdCross` => s""""$groupId" % "$artifactIdCross" % "$version""""
+      case _                 => s""""$groupId" %% "$artifactId" % "$version""""
```

Comment:
I see two problems here. One is that we assume that the dependency exists for the `scalaVersion` used in `ArtificialProject` which might not be true. And the other is that `%%` might cause problems with Scala.js artifacts.